src/core/dep-graph.ts

```diff
@@ -2,6 +2,12 @@ import * as _isEqual from 'lodash.isequal';
 import * as graphlib from '../graphlib';
 import * as types from './types';
 import { createFromJSON } from './create-from-json';
+import {
+  getMemoizedItem,
+  MemoizationMap,
+  memoize,
+} from '../legacy/memiozation';
```

Comment:
I haven't looked at the rest of this PR, but this is an issue – something from `core` mustn't have a dependency on something in `legacy`. Moving the memoization module also gives you an opportunity to fix the typo in its name :D

Reply:
Thanks, fixed the typo and moved it to another folder